src/Symfony/Component/DomCrawler/FormFieldRegistry.php

```diff
@@ -124,13 +124,15 @@ public function has($name)
     public function set($name, $value)
     {
         $target = &$this->get($name);
-        if (!is_array($value) || $target instanceof Field\ChoiceFormField) {
+        if ($target instanceof Field\FormField) {
```

Comment:
@jakzal I mean something like the following:

```php
$registry = new FormFieldRegistry();
$registry->add($this->getFormFieldMock('bar'));
$registry->set('bar', array('baz'));
```

Before, you got an error message like `InvalidArgumentException: Unreachable field "0"`. Now, the value will be implicitly cast to string. Is that really the desired behaviour?